### DISCUSSION TITLE
Warning if a variable is used in CALCULATE/CALCULATETABLE as the expression

### DISCUSSION THREAD

**Discussion by edhans** [TabularEditor/TabularEditor3#1261] [Ideas]
### Is your feature request related to a problem? Please describe

In much the same way you give a warning if a variable isn't used, it would be nice if you also gave a warning if a variable was used in an incorrect way, namely as the expression for CALCULATE

### Describe the solution you'd like

It is easy to get in this mode and I ran across this in another model today and I immediate recognized this is effectively doing nothing. 
<img width="564" alt="image" src="https://github.com/TabularEditor/TabularEditor3/assets/45298358/e1c947ea-f917-4c17-a60e-f6a1bc8df021">
Whatever varCoreMeasure is calculated as in the SWITCH() function, that is the value that is returned in the measure. The CALCULATE isn't doing anything.

### Goals

Just an aid to help even experienced developers that get wrapped up in a coding session and inadvertently do something like this.

### Additional context

I recognize you cannot protect us from everything, but some of the more obvious things just won't work and I think would be relatively simple to determine.

Where this would really help me as a manager is when I quickly scan models, I would immediately catch things like this without having to investigate every measure visually myself.

**Comment by edhans** [comment]
Additional note - 100% agnostic on any implementation. Yellow mark on the measure, a error in the messages window, and I'd be ok with a BPA rule if those can detect this sort of thing. I'd just need someone to write it. 😉

**Comment by mlonsk** [comment]
Hi @edhans 
This idea has been implemented as a Code Action in the Improvement category: https://docs.tabulareditor.com/te3/features/code-actions.html#:~:text=Remove%20superfluous%20CALCULATE
